### PR TITLE
§ Add support for sections

### LINKED ISSUE #60: Unrecognized section
After the conversion, the frame in the Figma section cannot be converted, and it is displayed as empty in the sketch file

### PR DESCRIPTION
**Pull request by esloho**
This PR adds basic support for sections which need to be handled the same way as frames in .fig files: if they hang directly under a canvas they're converted as artboards, and if they hang under another frame then they're converted as groups.

Fixes #60

### CODE REVIEW

File: converter/group.py [hunk 1]
```diff
@@ -18,7 +18,7 @@ def convert(fig_group):
 def post_process_frame(fig_group: dict, sketch_group: Group) -> Group:
     convert_frame_style(fig_group, sketch_group)
 
-    if fig_group["resizeToFit"]:
+    if "resizeToFit" in fig_group and fig_group["resizeToFit"]:
```
Comment (javitonino) [inline converter/group.py:21]: Why this change at all?

In case it's needed, then maybe do:

```suggestion
    if fig_group.get("resizeToFit"):
```

File: converter/tree.py [hunk 2]
```diff
@@ -83,8 +83,10 @@ def get_node_type(fig_node: dict, parent_type: str) -> str:
     # We do this because Sketch does not support nested artboards
     # If a Frame is detected inside another Frame, the internal one
     # is considered a group
-    if fig_node["type"] == "FRAME":
-        if parent_type == "CANVAS" and not fig_node["resizeToFit"]:
+    if fig_node["type"] == "FRAME" or fig_node["type"] == "SECTION":
+        if parent_type == "CANVAS" and (
+            "resizeToFit" not in fig_node or not fig_node["resizeToFit"]
```
Comment (javitonino) [inline converter/tree.py:88]:
```suggestion
            not fig_node.get("resizeToFit")
```

This also allow removing the wrapping parenthesis and maybe it fits in a single line

**Review by javitonino — COMMENTED** [review]
Some suggestions to further clean up code :broom:

File: converter/group.py (hunk 1 above)
Comment (esloho) [reply converter/group.py:21]: I got an error and assumed we don't have it for sections but I was wrong (it was just missing in my test). I'm undoing the change 👍

File: converter/tree.py (hunk 2 above)
Comment (esloho) [reply converter/tree.py:88]: https://github.com/sketch-hq/fig2sketch/pull/63#discussion_r1037933757

File: converter/tree.py [hunk 3]
```diff
@@ -83,8 +83,10 @@ def get_node_type(fig_node: dict, parent_type: str) -> str:
     # We do this because Sketch does not support nested artboards
     # If a Frame is detected inside another Frame, the internal one
     # is considered a group
-    if fig_node["type"] == "FRAME":
-        if parent_type == "CANVAS" and not fig_node["resizeToFit"]:
+    if fig_node["type"] == "FRAME" or fig_node["type"] == "SECTION":
```
[the remark this answers is not in the document]
Comment (esloho) [reply converter/tree.py]: 🤦‍♀️  https://github.com/sketch-hq/fig2sketch/pull/63/commits/9dca17ac0450198fe95f985c62058233d75a8de1

File: converter/base.py [hunk 4]
```diff
@@ -222,12 +222,12 @@ def resizing_constraint(fig_node: dict) -> int:
 
 
 def masking(fig_node: dict) -> _Masking:
-    CLIPPING_MODE = {
+    clipping_mode = {
```
[the remark this answers is not in the document]
Comment (esloho) [reply converter/base.py]: https://github.com/sketch-hq/fig2sketch/pull/63/commits/2b6ea4ab4861a90cd9790fa0914b14816a88fa70
